File: snippets/visualbasic/System/TimeSpan/Parse/parsefailure1.vb
' Visual Basic .NET Document
Option Strict On

Module Example
   Public Sub Main()
      ' <Snippet3>
      Dim values() As String = { "000000006", "12.12:12:12.12345678" }
      For Each value As String In values
         Try
            Dim interval As TimeSpan = TimeSpan.Parse(value)
            Console.WriteLine("{0} --> {1}", value, interval)
         Catch e As FormatException
            Console.WriteLine("{0}: Bad Format", value)
         Catch e As OverflowException
            Console.WriteLine("{0}: Overflow", value)
         End Try
      Next
      ' Output:
      '       000000006: Overflow
      '       12.12:12:12.12345678: Overflow
      ' </Snippet3>
   End Sub
End Module

File: snippets/fsharp/System/TimeSpan/Parse/parsefailure1.fs
module parsefailure1

// <Snippet3>
open System

let values = [| "000000006"; "12.12:12:12.12345678" |]
for value in values do
    try
        let interval = TimeSpan.Parse value
        printfn $"{value} --> {interval}"
    with
    | :? FormatException ->
        printfn $"{value}: Bad Format"
    | :? OverflowException ->
        printfn $"{value}: Overflow"

// Output:
//       000000006: Overflow
//       12.12:12:12.12345678: Overflow
// </Snippet3>
